src/NServiceBus.Core/Transports/TransportDefinition.cs

```diff
@@ -13,68 +13,110 @@ public abstract partial class TransportDefinition
+        /// <param name="logicalAddress">The logical address.</param>
+        /// <returns>The transport address.</returns>
+        public abstract string ToTransportAddress(LogicalAddress logicalAddress);
+
         /// <summary>
-        /// Returns the list of supported delivery constraints for this transport.
+        /// Returns the canonical for of the given transport address so various transport addresses can be effectively compared and deduplicated.
         /// </summary>
-        public abstract IEnumerable<Type> GetSupportedDeliveryConstraints();
+        /// <param name="transportAddress">A transport address.</param>
+        public virtual string MakeCanonicalForm(string transportAddress)
+        {
+            return transportAddress;
+        }
 
         /// <summary>
-        /// Gets the highest supported transaction mode for the this transport.
+        /// Initializes all the factories and supported features for the transport. This method is called right before all features are activated and the settings will be locked down. This means you can use the SettingsHolder both for providing default capabilities as well as for initializing the transport's configuration based on those settings (the user cannot provide information anymore at this stage).
         /// </summary>
-        public abstract TransportTransactionMode GetSupportedTransactionMode();
+        /// <param name="settings">An instance of the current settings.</param>
+        /// <returns>The supported factories.</returns>
+        protected abstract TransportInfrastructure Initialize(SettingsHolder settings);
 
         /// <summary>
-        /// Will be called if the transport has indicated that it has native support for pub sub.
-        /// Creates a transport address for the input queue defined by a logical address.
+        /// Gets an example connection string to use when reporting lack of configured connection string to the user.
         /// </summary>
-        public abstract IManageSubscriptions GetSubscriptionManager();
+        public abstract string ExampleConnectionStringForErrorMessage { get; }
 
         /// <summary>
-        /// Returns the discriminator for this endpoint instance.
+        /// Used by implementations to control if a connection string is necessary.
         /// </summary>
-        public abstract EndpointInstance BindToLocalEndpoint(EndpointInstance instance, ReadOnlySettings settings);
+        public virtual bool RequiresConnectionString => true;
+
 
         /// <summary>
-        /// Converts a given logical address to the transport address.
+        /// The factories &amp; support infrastructure for this transport.
         /// </summary>
-        /// <param name="logicalAddress">The logical address.</param>
-        /// <returns>The transport address.</returns>
-        public abstract string ToTransportAddress(LogicalAddress logicalAddress);
+        public TransportInfrastructure Infrastructure { get; private set; }
```

Comment:
What I think could work is changing current `TransportDefinition` to `TransportInfrastructureBuilder` with single method Build which gets settings. This method would return `TransportInfrastructure` with all the methods that it has right now plus any which is now on the `TransprotDefinition`.